collect_app/src/main/java/org/odk/collect/android/preferences/ResetDialogPreference.java

```diff
@@ -181,24 +181,7 @@ private void handleResult(List<Integer> resetActions, List<Integer> failedResetA
     }
 
     private void showResultDialog(final String resultMessage) {
-        ((AdminPreferencesActivity) context).runOnUiThread(new Runnable() {
-            @Override
-            public void run() {
-                AlertDialog.Builder b = new AlertDialog.Builder(getContext());
-                b.setTitle(getContext().getString(R.string.reset_app_state_result));
-                b.setMessage(resultMessage);
-                b.setCancelable(false);
-                b.setIcon(android.R.drawable.ic_dialog_info);
-                b.setPositiveButton(getContext().getString(R.string.ok), new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        dialog.dismiss();
-                        ((AdminPreferencesActivity) context).recreate();
-                    }
-                });
-                AlertDialog alertDialog = b.create();
-                alertDialog.show();
-            }
-        });
+        SimpleDialog simpleDialog = SimpleDialog.newInstance(getContext().getString(R.string.reset_app_state_result), android.R.drawable.ic_dialog_info, resultMessage, getContext().getString(R.string.ok));
```

Comment:
I do think this is hard to read. I would write it as 
```
String title = SimpleDialog.newInstance(getContext().getString(R.string.reset_app_state_result);
int iconID = android.R.drawable.ic_dialog_info;
String buttonTitle = getContext().getString(R.string.ok);
SimpleDialog simpleDialog = SimpleDialog.newInstance(title, iconID, resultMessage, buttonTitle);
```

That more closely matches what you liked about using a `Builder`, @grzesiek2010.